File: conversions/repeat-string/elide-repeat-string.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-repeat-string

This demonstrates calling the TypeScript repeat-string implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One repeat-string implementation shared across TypeScript and Python
- Consistent repeat a string n times efficiently across services
- No Python custom implementation needed
- Perfect for text formatting, padding, ASCII art, UI borders
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# repeat_string_module = require('./elide-repeat-string.ts')

print("=== Python Calling TypeScript repeat-string ===\n")

# Example: Basic usage
# result = repeat_string_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses repeat a string n times efficiently")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for text formatting, padding, ASCII art, UI borders")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide repeat-string (TypeScript)         │")
print("│   conversions/repeat-string/elide-repeat-string.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same repeat a string n times efficiently everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  repeat_string = require('./elide-repeat-string.ts')")
print("  # Use it directly!")

File: converted/utilities/repeat-string/elide-repeat-string.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-repeat-string
#
# This demonstrates calling the TypeScript repeat-string implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One repeat-string implementation shared across TypeScript and Ruby
# - Consistent repeat a string n times efficiently across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript repeat-string ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# repeat_string_module = Elide.require('./elide-repeat-string.ts')

# Example: Basic usage
# result = repeat_string_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses repeat a string n times efficiently"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for text formatting, padding, ASCII art, UI borders"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide repeat-string (TypeScript)         │"
puts "│   conversions/repeat-string/elide-repeat-string.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same repeat a string n times efficiently everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/repeat_string.rb"
puts "  REPEAT_STRING = Elide.require('./elide-repeat-string.ts')"
